api/v3/Mailing.php

```diff
@@ -575,11 +575,21 @@ function civicrm_api3_mailing_preview($params) {
   if (!$contactID) {
     $contactID = $session->get('userID');
   }
+  // If we still don't have a contact id because the previous call returned NULL set contactID to be 0
+  if (!$contactID) {
```

Comment:
@seamuslee001 on the form layer we use a function getContactID that does these things & also checks for a cs - less relevant here maybe - I wonder if we can re-use that?

Otherwise we can maybe make it tidier with a ternary

Reply:
@eileenmcnaughton have simplified it down